app/build.gradle.kts

```diff
@@ -0,0 +1,172 @@
+
+        ksp {
+            arg("room.incremental","true")
+        }
+    }
+
+    flavorDimensions.add("channel")
+    productFlavors {
+        create("github") {
+            dimension = "channel"
+        }
+        create("fdroid") {
+            dimension = "channel"
+        }
+        create("googlePlay") {
+            dimension = "channel"
+            applicationIdSuffix = ".google.play"
+        }
+    }
+    signingConfigs {
+        create("release") {
+            keyAlias = keyProps["keyAlias"] as String?
+            keyPassword = keyProps["keyPassword"] as String?
+            storeFile = keyProps["storeFile"]?.let { file(it as String) }
+            storePassword = keyProps["storePassword"] as String?
+        }
+    }
+    lint {
+        disable.addAll(listOf("MissingTranslation", "ExtraTranslation"))
+    }
+    buildTypes {
+        getByName("release") {
+            isMinifyEnabled = true
+            isShrinkResources = true
+            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
+            signingConfig = signingConfigs.getByName("release")
+        }
+    }
+    applicationVariants.all {
+        outputs.all {
+            (this as com.android.build.gradle.internal.api.BaseVariantOutputImpl).outputFileName = "ReadYou-${defaultConfig.versionName}-${gitCommitHash}.apk"
+        }
+    }
+    kotlinOptions {
+        freeCompilerArgs = freeCompilerArgs + "-opt-in=kotlin.RequiresOptIn"
+        jvmTarget = JavaVersion.VERSION_11.toString()
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
+    }
+    buildFeatures {
+        compose = true
+        buildConfig = true
+    }
+    composeOptions {
+        kotlinCompilerExtensionVersion = libs.versions.composeCompiler.get()
+    }
+    packaging {
+        resources.excludes.add("/META-INF/{AL2.0,LGPL2.1}")
+    }
+    androidResources {
+        generateLocaleConfig = true
+    }
+    namespace = "me.ash.reader"
+}
+
+dependencies {
+    implementation(libs.aboutLibrariesCore)
+    implementation(libs.aboutLibrariesComposeM3)
+    implementation(libs.composeHtml)
+    implementation(libs.androidSVG)
+    implementation(libs.opmlParser)
+    implementation(libs.readability4j)
+    implementation(libs.rome)
+    implementation(libs.coilBase)
+    implementation(libs.coilCompose)
+    implementation(libs.coilSvg)
+    implementation(libs.coilGif)
+    implementation(libs.telephoto)
+    implementation(libs.okhttp)
+    implementation(libs.okhttpCoroutines)
+    implementation(libs.retrofit)
+    implementation(libs.retrofitGson)
+    implementation(libs.profileinstaller)
+    implementation(libs.workRuntimeKtx)
+    implementation(libs.datastorePreferences)
+    implementation(libs.roomPaging)
+    implementation(libs.roomCommon)
+    implementation(libs.roomKtx)
+    ksp(libs.roomCompiler)
+    implementation(libs.pagingCommonKtx)
```

Comment:
Btw `libs.xxxxxx` seems not very well-structure, formats like `libs.coil.svg` and `libs.coil.compose` are preferred

Reply:
reference: https://github.com/android/nowinandroid/blob/main/gradle/libs.versions.toml